<!DOCTYPE html>

<html lang="en" data-content_root="./">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>hichiarep package &#8212; HiChIA-rep 0.1.dev62+g81d4e8980.d20251218 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=5ecbeea2" />
    <link rel="stylesheet" type="text/css" href="_static/basic.css?v=686e5160" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css?v=27fed22d" />
    <script src="_static/documentation_options.js?v=35e7d822"></script>
    <script src="_static/doctools.js?v=9bcbadda"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Welcome to HiChIA-Rep’s documentation!" href="index.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  

  
  

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="hichiarep-package">
<h1>hichiarep package<a class="headerlink" href="#hichiarep-package" title="Link to this heading">¶</a></h1>
<section id="module-hichiarep.chrom_bin_data">
<span id="hichiarep-chrom-bin-data-module"></span><h2>hichiarep.chrom_bin_data module<a class="headerlink" href="#module-hichiarep.chrom_bin_data" title="Link to this heading">¶</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.emd">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chrom_bin_data.</span></span><span class="sig-name descname"><span class="pre">emd</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">ndarray</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">q</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">ndarray</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Tuple</span><span class="p"><span class="pre">[</span></span><span class="pre">float</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">float</span><span class="p"><span class="pre">]</span></span></span></span><a class="headerlink" href="#hichiarep.chrom_bin_data.emd" title="Link to this definition">¶</a></dt>
<dd><p>LEGACY CODE
Finds the Earth Mover’s Distance of two 1D arrays</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>p</strong> (<em>ndarray</em>)</p></li>
<li><p><strong>q</strong> (<em>ndarray</em>)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The Earth Mover’s Distance; the weight of this calculation</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>tuple[float, float]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.jensen_shannon_divergence">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chrom_bin_data.</span></span><span class="sig-name descname"><span class="pre">jensen_shannon_divergence</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">ndarray</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">q</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">ndarray</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">2</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">float</span></span></span><a class="headerlink" href="#hichiarep.chrom_bin_data.jensen_shannon_divergence" title="Link to this definition">¶</a></dt>
<dd><p>LEGACY CODE
Finds the jensen-shannon divergence</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>p</strong> (<em>np.ndarray</em><em>[</em><em>np.float64</em><em>]</em>) – Graph of sample1</p></li>
<li><p><strong>q</strong> (<em>np.ndarray</em><em>[</em><em>np.float64</em><em>]</em>) – Graph of sample2</p></li>
<li><p><strong>base</strong> (<em>int</em><em>, </em><em>optional</em>) – Determines base to be used in calculating scipy.entropy (Default is 2)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Jensen-Shannon divergence</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>float</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.output_graph">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chrom_bin_data.</span></span><span class="sig-name descname"><span class="pre">output_graph</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">n</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">parent_dir</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">do_output</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sparse</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.output_graph" title="Link to this definition">¶</a></dt>
<dd><p>Outputs the graphs as images for visual inspection</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.construct_windows">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chrom_bin_data.</span></span><span class="sig-name descname"><span class="pre">construct_windows</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">chrom_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_stride</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bin_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.construct_windows" title="Link to this definition">¶</a></dt>
<dd><p>Computes the window indices and locations for this chromosome</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>chrom_size</strong> (<em>int</em>) – The size of the chromosome</p></li>
<li><p><strong>window_size</strong> (<em>int</em>) – Size of sliding window</p></li>
<li><p><strong>window_stride</strong> (<em>int</em>) – Factor of window_size to stride (e.g. 2 means stride by window_size/2)</p></li>
<li><p><strong>bin_size</strong> (<em>int</em>) – Binning resolution</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><dl class="simple">
<dt>window_indices<span class="classifier">np.ndarray</span></dt><dd><p>A Mx2 array where M is the number of windows
The start and end indices of each window in bin units</p>
</dd>
<dt>window_locations<span class="classifier">np.ndarray</span></dt><dd><p>A Mx2 array where M is the number of windows
The start and end locations of each window in base pairs</p>
</dd>
</dl>
</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>tuple[np.ndarray, np.ndarray]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.apply_mean_filter">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chrom_bin_data.</span></span><span class="sig-name descname"><span class="pre">apply_mean_filter</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">A</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ks</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">3</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.apply_mean_filter" title="Link to this definition">¶</a></dt>
<dd><p>Helper function for GSP methods</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.pmf">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chrom_bin_data.</span></span><span class="sig-name descname"><span class="pre">pmf</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">x</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.pmf" title="Link to this definition">¶</a></dt>
<dd><p>Helper function for GSP methods</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.random_walk">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chrom_bin_data.</span></span><span class="sig-name descname"><span class="pre">random_walk</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">A</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">p</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.random_walk" title="Link to this definition">¶</a></dt>
<dd><p>Helper function for GSP methods</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.compare_signals">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chrom_bin_data.</span></span><span class="sig-name descname"><span class="pre">compare_signals</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">a</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">b</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">compare_method</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.compare_signals" title="Link to this definition">¶</a></dt>
<dd><p>Helper function for GSP methods</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.laplacian_sparse">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chrom_bin_data.</span></span><span class="sig-name descname"><span class="pre">laplacian_sparse</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">A</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">laplacian_type</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.laplacian_sparse" title="Link to this definition">¶</a></dt>
<dd><p>Helper function for GSP methods
A is a CSR matrix (deprecated)</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.laplacian">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chrom_bin_data.</span></span><span class="sig-name descname"><span class="pre">laplacian</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">A</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">laplacian_type</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.laplacian" title="Link to this definition">¶</a></dt>
<dd><p>Helper function for GSP methods
A is a dense matrix</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.construct_diffusion_kernel">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chrom_bin_data.</span></span><span class="sig-name descname"><span class="pre">construct_diffusion_kernel</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">L</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">t</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.construct_diffusion_kernel" title="Link to this definition">¶</a></dt>
<dd><p>Helper function for GSP methods</p>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">hichiarep.chrom_bin_data.</span></span><span class="sig-name descname"><span class="pre">ChromBinData</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">chrom_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sample_name</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bin_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_stride</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData" title="Link to this definition">¶</a></dt>
<dd><p>A class used to represent a chromosome in a sample</p>
<p>The difference with <cite>ChromLoopData</cite> class is that this class is a binned representation
of loops, as opposed to loop-level granularity</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.name">
<span class="sig-name descname"><span class="pre">name</span></span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.name" title="Link to this definition">¶</a></dt>
<dd><p>The name of the chromosome</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.size">
<span class="sig-name descname"><span class="pre">size</span></span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.size" title="Link to this definition">¶</a></dt>
<dd><p>The size of the chromosome</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>int</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.sample_name">
<span class="sig-name descname"><span class="pre">sample_name</span></span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.sample_name" title="Link to this definition">¶</a></dt>
<dd><p>The name of the sample this chromosome belongs to</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.bin_size">
<span class="sig-name descname"><span class="pre">bin_size</span></span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.bin_size" title="Link to this definition">¶</a></dt>
<dd><p>The size of each bin i.e. resolution</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>int</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.window_indices">
<span class="sig-name descname"><span class="pre">window_indices</span></span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.window_indices" title="Link to this definition">¶</a></dt>
<dd><p>The start and end indices of each window in bin units</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>np.ndarray</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.window_locations">
<span class="sig-name descname"><span class="pre">window_locations</span></span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.window_locations" title="Link to this definition">¶</a></dt>
<dd><p>The start and end locations of each window in base pairs</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>np.ndarray</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.adjacency_matrices">
<span class="sig-name descname"><span class="pre">adjacency_matrices</span></span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.adjacency_matrices" title="Link to this definition">¶</a></dt>
<dd><p>A list of NxN COO sparse adjacency matrices for each window.
N is the number of bins per window.</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>list</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.node_weights">
<span class="sig-name descname"><span class="pre">node_weights</span></span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.node_weights" title="Link to this definition">¶</a></dt>
<dd><p>A MxN array of binding affinity for the chromosome
M is the number of windows
N is the number of bins per window</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>np.ndarray</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.finish_init">
<span class="sig-name descname"><span class="pre">finish_init</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.finish_init" title="Link to this definition">¶</a></dt>
<dd><p>Finalize the initialization of this chromosome data
Checks the sizes of each attributes. Specifically that</p>
<blockquote>
<div><ul class="simple">
<li><p>self.adjacency_matrices is a list of M sparse COO matrices that are (N, N)</p></li>
<li><p>self.node_weights is shape (M, N)</p></li>
</ul>
</div></blockquote>
<p>where M is the number of windows and N is the number of bins per window</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.filter_with_peaks">
<span class="sig-name descname"><span class="pre">filter_with_peaks</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">peak_array</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">ndarray</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">both_peak_support</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">bool</span></span></span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.filter_with_peaks" title="Link to this definition">¶</a></dt>
<dd><p>Filters out Hi-C contacts without peak support using matrix operations</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>peak_array</strong> (<em>np.ndarray</em>) – 1D array of peak values (dense)</p></li>
<li><p><strong>both_peak_support</strong> (<em>bool</em><em>, </em><em>optional</em>) – Whether to only keep loops that have peak support on both sides
(default is False)</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Whether the chromosome had any problems when filtering</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>bool</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.fgw_distance_batch">
<span class="sig-name descname"><span class="pre">fgw_distance_batch</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">g1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">g2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">x2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alpha</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cost</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gvmax</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">100.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scale_cost</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'unitless'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mass</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_cores</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.fgw_distance_batch" title="Link to this definition">¶</a></dt>
<dd><p>DEPRECATED</p>
<p>Computes the FGW on batches of attributed graphs
Specifically, we batch across the first dimension (M):</p>
<blockquote>
<div><ul class="simple">
<li><p>g1: MxNxN</p></li>
<li><p>x1: MxN</p></li>
</ul>
</div></blockquote>
<p>Returns a list of FGW distances for each pair in the batch and the window weights</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.gsp_batch">
<span class="sig-name descname"><span class="pre">gsp_batch</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">o_chrom</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">kernel_type</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mu</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">compare_method</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cross</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_cores</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.gsp_batch" title="Link to this definition">¶</a></dt>
<dd><p>Computes the GSP distances on batches of graphs.
Uses sparse matrices directly for efficiency with expm_multiply.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>o_chrom</strong> (<a class="reference internal" href="#hichiarep.chrom_bin_data.ChromBinData" title="hichiarep.chrom_bin_data.ChromBinData"><em>ChromBinData</em></a>) – The other chromosome to compare against</p></li>
<li><p><strong>kernel_type</strong> (<em>str</em>) – Type of graph kernel to use: either “diffusion” or “random_walk”</p></li>
<li><p><strong>mu</strong> (<em>float</em><em> or </em><em>int</em>) – Diffusion time parameter
OR
Power for random walk</p></li>
<li><p><strong>compare_method</strong> (<em>str</em>) – Method to compare diffusion states (e.g., “spearman”, “jsd”)</p></li>
<li><p><strong>num_cores</strong> (<em>int</em>) – Number of cores to use</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.wasserstein_batch">
<span class="sig-name descname"><span class="pre">wasserstein_batch</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">o_chrom</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">feat</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weight</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_cores</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.wasserstein_batch" title="Link to this definition">¶</a></dt>
<dd><p>DEPRECATED</p>
<p>Computes the Wasserstein distance on batches of graphs.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>o_chrom</strong> (<a class="reference internal" href="#hichiarep.chrom_bin_data.ChromBinData" title="hichiarep.chrom_bin_data.ChromBinData"><em>ChromBinData</em></a>) – The other chromosome to compare against</p></li>
<li><p><strong>num_cores</strong> (<em>int</em>) – Number of cores to use</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.emd_jsd_batch">
<span class="sig-name descname"><span class="pre">emd_jsd_batch</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">o_chrom</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">method</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_cores</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.emd_jsd_batch" title="Link to this definition">¶</a></dt>
<dd><p>Computes JSD or EMD on batches of graphs using the old logic.</p>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="hichiarep.chrom_bin_data.ChromBinData.compare">
<span class="sig-name descname"><span class="pre">compare</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">o_chrom</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#hichiarep.chrom_bin_data.ChromBinData" title="hichiarep.chrom_bin_data.ChromBinData"><span class="pre">ChromBinData</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">alpha</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">method</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'random_walk'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mu</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cross</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">compare_method</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'spearman'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cost</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'linear'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'output'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">do_output_graph</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">param_str</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">''</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gvmax</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">100.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scale_cost</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'unitless'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mass</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">feat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'index_BA'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weight</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'uniform'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_cores</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chrom_bin_data.ChromBinData.compare" title="Link to this definition">¶</a></dt>
<dd><p>Compares this chromosome with another chromosome using the specified method</p>
<p>The sparse matrices are all converted to dense matrices for batch comparison.
Specifically:</p>
<ol class="arabic simple">
<li><p>Adjacency matrices: MxNxN</p></li>
<li><p>Node weights: MxN</p></li>
</ol>
<p>where M is the number of windows and N is the number of bins per window</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>o_chrom</strong> (<a class="reference internal" href="#hichiarep.chrom_bin_data.ChromBinData" title="hichiarep.chrom_bin_data.ChromBinData"><em>ChromBinData</em></a>) – The other chromosome to compare against</p></li>
<li><p><strong>method</strong> (<em>str</em>) – <p>Comparison method to use. Options are:</p>
<ul>
<li><p>’random_walk’</p></li>
<li><p>’diffusion’</p></li>
</ul>
</p></li>
<li><p><strong>mu</strong> (<em>float</em><em> or </em><em>int</em>) – Power to raise the random walk transition matrix (if method is ‘random_walk’)
Diffusion time step (if method is ‘diffusion’)</p></li>
<li><p><strong>cross</strong> (<em>bool</em>) – Whether to compare the signals ‘direct’ or ‘cross’.
See chia_rep.compare() documentation for details.</p></li>
<li><p><strong>compare_method</strong> (<em>str</em>) – Method to compare processed binding affinity signals
after passing through random walk or diffusion.
Either ‘spearman’ or ‘jsd’</p></li>
<li><p><strong>output_dir</strong> (<em>str</em>) – Output folder</p></li>
<li><p><strong>do_output_graph</strong> (<em>bool</em>) – Whether to output the processed graphs for each window of chromosome 1
as .npys for debugging purposes</p></li>
<li><p><strong>num_cores</strong> (<em>int</em>) – Number of pools to use for parallel processing across the windows of a chromosome</p></li>
<li><p><strong>param_str</strong> (<em>str</em>) – Parameter string that is the subfolder name under output_dir</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</dd></dl>

</section>
<section id="module-hichiarep.genome_bin_data">
<span id="hichiarep-genome-bin-data-module"></span><h2>hichiarep.genome_bin_data module<a class="headerlink" href="#module-hichiarep.genome_bin_data" title="Link to this heading">¶</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.genome_bin_data.nan_average">
<span class="sig-prename descclassname"><span class="pre">hichiarep.genome_bin_data.</span></span><span class="sig-name descname"><span class="pre">nan_average</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">values</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weights</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.genome_bin_data.nan_average" title="Link to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.genome_bin_data.check_alt_chrom_name">
<span class="sig-prename descclassname"><span class="pre">hichiarep.genome_bin_data.</span></span><span class="sig-name descname"><span class="pre">check_alt_chrom_name</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">chrom_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chromosome_list</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.genome_bin_data.check_alt_chrom_name" title="Link to this definition">¶</a></dt>
<dd><p>Checks for match of chrom_name in chromosome_list</p>
<p>Considers alternate chromosome names (chr1 vs 1)</p>
<p>Returns None if not found. Otherwise returns the matched chromosome name</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>chrom_name</strong> (<em>str</em>) – Chromosome name to check</p></li>
<li><p><strong>chromosome_list</strong> (<em>list</em>) – List of chromosome names to check against</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.genome_bin_data.weigh_adjacency_matrix_binding_affinity">
<span class="sig-prename descclassname"><span class="pre">hichiarep.genome_bin_data.</span></span><span class="sig-name descname"><span class="pre">weigh_adjacency_matrix_binding_affinity</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">A</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">b</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.genome_bin_data.weigh_adjacency_matrix_binding_affinity" title="Link to this definition">¶</a></dt>
<dd><p>Weighs the batch adjacency matrix (MxNxN) by the binding affinity node weights (MxN)
according to the rule:
A[:,i,j] = A[:,i,j] * (b[:,i] + b[:,j])</p>
<p>The first dimension is the window index and is simply broadcasted over</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>A</strong> (<em>np.ndarray</em>) – MxNxN array of adjacency matrices</p></li>
<li><p><strong>b</strong> (<em>np.ndarray</em>) – MxN array of binding affinity node weights</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><strong>out</strong> – MxNxN array of weighed adjacency matrices</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>np.ndarray</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.genome_bin_data.threshold_peaks">
<span class="sig-prename descclassname"><span class="pre">hichiarep.genome_bin_data.</span></span><span class="sig-name descname"><span class="pre">threshold_peaks</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">p_array</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_peaks</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">total_peaks</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">to_remove</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom_name</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_ratio</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.genome_bin_data.threshold_peaks" title="Link to this definition">¶</a></dt>
<dd><p>Thresholds the peaks, assuming that <cite>p_array</cite> is a 1D numpy array of peak values</p>
<p>Updates <cite>to_remove</cite> list for chromosomes that need to be removed</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>p_array</strong> (<em>np.ndarray</em>) – 1D numpy array of peak values</p></li>
<li><p><strong>num_peaks</strong> (<em>int</em>) – Used by this function to simply determine if we need to threshold
i.e. if num_peaks is None, then no thresholding</p></li>
<li><p><strong>total_peaks</strong> (<em>int</em>) – Total number of peaks in this chromosome</p></li>
<li><p><strong>to_remove</strong> (<em>list</em>) – List of chromosome names to remove</p></li>
<li><p><strong>chrom_name</strong> (<em>str</em>) – Name of the chromosome being processed</p></li>
<li><p><strong>base_ratio</strong> (<em>float</em>) – The ratio of peaks to keep based on base chromosome</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p><ul class="simple">
<li><p><strong>p_array_thresh</strong> (<em>np.ndarray</em>) – Thresholded peak array</p></li>
<li><p><strong>to_remove</strong> (<em>list</em>) – Updated list of chromosome names to remove</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.genome_bin_data.read_bedpe_and_bin">
<span class="sig-prename descclassname"><span class="pre">hichiarep.genome_bin_data.</span></span><span class="sig-name descname"><span class="pre">read_bedpe_and_bin</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bedpe_file</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bin_size</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_hic_value</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.genome_bin_data.read_bedpe_and_bin" title="Link to this definition">¶</a></dt>
<dd><p>Read in Hi-C bedpe file and bin the interactions according to bin_size</p>
<p>Code is almost identical to the legacy code in GenomeLoopData with a few extra checks</p>
<dl class="field-list simple">
<dt class="field-odd">Returns<span class="colon">:</span></dt>
<dd class="field-odd"><p><ul class="simple">
<li><p><strong>bedpe_data</strong> (<em>dict</em>) – Key: chrom name
Value: np.array of shape (num_interactions, 3) where each row is (b1, b2, val)
b1 and b2 are the binned positions of the two anchors
val is the interaction value</p></li>
<li><p><strong>hic_chromosomes</strong> (<em>list</em>) – List of chromosomes found in the bedpe file</p></li>
</ul>
</p>
</dd>
</dl>
</dd></dl>

<dl class="py class">
<dt class="sig sig-object py" id="hichiarep.genome_bin_data.GenomeBinData">
<em class="property"><span class="pre">class</span><span class="w"> </span></em><span class="sig-prename descclassname"><span class="pre">hichiarep.genome_bin_data.</span></span><span class="sig-name descname"><span class="pre">GenomeBinData</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">chrom_size_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bedgraph_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom_structure_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_stride</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">normalization</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'NONE'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">peak_dict</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">list</span><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">from_peak_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_chrom</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'chr1'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_peaks</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chroms_to_load</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bin_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_hic_value</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_bedgraph_value</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ba_mult</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.genome_bin_data.GenomeBinData" title="Link to this definition">¶</a></dt>
<dd><p>A class used to represent a sample</p>
<dl class="py attribute">
<dt class="sig sig-object py" id="hichiarep.genome_bin_data.GenomeBinData.species_name">
<span class="sig-name descname"><span class="pre">species_name</span></span><a class="headerlink" href="#hichiarep.genome_bin_data.GenomeBinData.species_name" title="Link to this definition">¶</a></dt>
<dd><p>Name of the sample’s species (hg38, mm10, …)</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="hichiarep.genome_bin_data.GenomeBinData.sample_name">
<span class="sig-name descname"><span class="pre">sample_name</span></span><a class="headerlink" href="#hichiarep.genome_bin_data.GenomeBinData.sample_name" title="Link to this definition">¶</a></dt>
<dd><p>Name of the sample (LHH0061, LHH0061_0061H, …)</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>str</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="hichiarep.genome_bin_data.GenomeBinData.chrom_dict">
<span class="sig-name descname"><span class="pre">chrom_dict</span></span><a class="headerlink" href="#hichiarep.genome_bin_data.GenomeBinData.chrom_dict" title="Link to this definition">¶</a></dt>
<dd><p>Key: Name of chromosome
Value: ChromBinData object</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict[str, <a class="reference internal" href="#hichiarep.chrom_bin_data.ChromBinData" title="hichiarep.chrom_bin_data.ChromBinData">ChromBinData</a>]</p>
</dd>
</dl>
</dd></dl>

<dl class="py attribute">
<dt class="sig sig-object py" id="hichiarep.genome_bin_data.GenomeBinData.peak_dict">
<span class="sig-name descname"><span class="pre">peak_dict</span></span><a class="headerlink" href="#hichiarep.genome_bin_data.GenomeBinData.peak_dict" title="Link to this definition">¶</a></dt>
<dd><p>Key: Name of chromosome
Value: 1D numpy array of number of bins in chromosome.
Each entry represents the peak value for that bin</p>
<dl class="field-list simple">
<dt class="field-odd">Type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict[str, np.ndarray]</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="hichiarep.genome_bin_data.GenomeBinData.preprocess">
<span class="sig-name descname"><span class="pre">preprocess</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">both_peak_support</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'output'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">diagnostic_plots</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#hichiarep.genome_bin_data.GenomeBinData.preprocess" title="Link to this definition">¶</a></dt>
<dd><p>Filters out Hi-C contacts without peak support using matrix operations</p>
<p>Removes chromosomes that result in all zero matrices after filtering</p>
<p>Optionally saves diagnostic plots of peak array</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>both_peak_support</strong> (<em>bool</em><em>, </em><em>optional</em>) – Whether to only keep loops that have peak support on both sides</p></li>
<li><p><strong>output_dir</strong> (<em>str</em><em>, </em><em>optional</em>) – Directory to output found peaks and filters</p></li>
<li><p><strong>diagnostic_plots</strong> (<em>bool</em><em>, </em><em>optional</em>) – Whether to save diagnostic plots of peak array</p></li>
</ul>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt class="sig sig-object py" id="hichiarep.genome_bin_data.GenomeBinData.compare">
<span class="sig-name descname"><span class="pre">compare</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">o_loop_data</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><a class="reference internal" href="#hichiarep.genome_bin_data.GenomeBinData" title="hichiarep.genome_bin_data.GenomeBinData"><span class="pre">GenomeBinData</span></a></span></em>, <em class="sig-param"><span class="n"><span class="pre">alpha</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">method</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'fgw'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cost</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'linear'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chroms_to_compare</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'output'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">do_output_graph</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gvmax</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">100.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scale_cost</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'unitless'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mass</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">feat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'index_BA'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weight</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'uniform'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mu</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">compare_method</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'spearman'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cross</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_cores</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">param_str</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">''</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">float</span><span class="p"><span class="pre">]</span></span></span></span><a class="headerlink" href="#hichiarep.genome_bin_data.GenomeBinData.compare" title="Link to this definition">¶</a></dt>
<dd><p>Compares this sample (genome bin data) to another sample (genome bin data)</p>
<p>Gets the comparison values for each window in each chromosome and
combines that into a genome-wide comparison value. Each window is given
a weight based on the highest loop in it. Each chromosome is weighted
equally</p>
</dd></dl>

</dd></dl>

</section>
<section id="module-hichiarep.util">
<span id="hichiarep-util-module"></span><h2>hichiarep.util module<a class="headerlink" href="#module-hichiarep.util" title="Link to this heading">¶</a></h2>
</section>
<section id="module-hichiarep.chia_rep">
<span id="hichiarep-chia-rep-module"></span><h2>hichiarep.chia_rep module<a class="headerlink" href="#module-hichiarep.chia_rep" title="Link to this heading">¶</a></h2>
<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chia_rep.output_score">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chia_rep.</span></span><span class="sig-name descname"><span class="pre">output_score</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">scores</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">file_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#hichiarep.chia_rep.output_score" title="Link to this definition">¶</a></dt>
<dd><p>Output scores to a specified file path.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>scores</strong></p></li>
<li><p><strong>file_path</strong></p></li>
</ul>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chia_rep.output_to_csv">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chia_rep.</span></span><span class="sig-name descname"><span class="pre">output_to_csv</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">weighted_scores</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">unweighted_scores</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">float</span><span class="p"><span class="pre">]</span></span><span class="p"><span class="pre">]</span></span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'output'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">param_str</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#hichiarep.chia_rep.output_to_csv" title="Link to this definition">¶</a></dt>
<dd><p>Outputs weighted and unweighted scores to a specified file path.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>weighted_scores</strong></p></li>
<li><p><strong>unweighted_scores</strong></p></li>
<li><p><strong>output_dir</strong> – Directory to output scores</p></li>
</ul>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chia_rep.compare">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chia_rep.</span></span><span class="sig-name descname"><span class="pre">compare</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_dict</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">OrderedDict</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">method</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'random_walk'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cost</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'linear'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">compare_list</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">list</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">compare_list_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'output'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">alpha</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">0.5</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">do_output_graph</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gvmax</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">100.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">scale_cost</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'unitless'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mass</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">feat</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'index_BA'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weight</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'uniform'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">mu</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">float</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1.0</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">compare_method</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'spearman'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cross</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_cores</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">param_str</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chia_rep.compare" title="Link to this definition">¶</a></dt>
<dd><p>Compares specified samples against each other. Specify comparisons in either
compare_list or compare_list_file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sample_dict</strong> (<em>OrderedDict</em>) – (Key: sample name, value: sample data)</p></li>
<li><p><strong>method</strong> (<em>str</em>) – <p>Comparison method to use. Options are:</p>
<ul>
<li><p>’random_walk’</p></li>
<li><p>’diffusion’</p></li>
</ul>
<p>Deprecated options:</p>
<ul>
<li><dl class="simple">
<dt>’w’ (wasserstein)</dt><dd><p>Associated parameters: <cite>feat</cite>, <cite>weight</cite></p>
</dd>
</dl>
</li>
<li><dl class="simple">
<dt>’fgw’ (fused gromov-wasserstein)</dt><dd><p>Associated parameters: <cite>alpha</cite>, <cite>cost</cite>, <cite>gvmax</cite>, <cite>scale_cost</cite>, <cite>mass</cite></p>
</dd>
</dl>
</li>
<li><p>’JSD’ (legacy JSD)</p></li>
<li><p>’EMD’ (legacy EMD)</p></li>
</ul>
</p></li>
<li><p><strong>compare_list</strong> (<em>list</em><em>, </em><em>optional</em>) – List of comparisons to make. Shape is n x 2 where n is the number of
comparisons</p></li>
<li><p><strong>compare_list_file</strong> (<em>str</em><em>, </em><em>optional</em>) – File that contains a list of comparisons to make.
Format:
sample1_name    sample2_name
sample3_name    sample4_name
…</p></li>
<li><p><strong>output_dir</strong> (<em>str</em>) – Directory to output data</p></li>
<li><p><strong>do_output_graph</strong> (<em>bool</em>) – Output processed .npy graphs for each window of chromosome 1 only
This is mainly for developer debugging purposes</p></li>
<li><p><strong>mu</strong> (<em>float</em><em> or </em><em>int</em>) – Power to raise the random walk transition matrix (if method is ‘random_walk’)
Diffusion time step (if method is ‘diffusion’)</p></li>
<li><p><strong>compare_method</strong> (<em>str</em>) – Method to compare processed binding affinity signals
after passing through random walk or diffusion.
Either ‘spearman’ or ‘jsd’</p></li>
<li><p><strong>cross</strong> (<em>bool</em>) – <p>Whether to compare the signals ‘direct’ or ‘cross’.</p>
<p>Let (x1, K1) be the binding affinity and linear operator for sample 1
Let (x2, K2) be the binding affinity and linear operator for sample 2</p>
<p>Then,</p>
<p>Direct comparison:
x1 -&gt; K1 -&gt; x11
x2 -&gt; K2 -&gt; x22
Compare x11 and x22</p>
<p>Cross comparison:
x1 -&gt; K2 -&gt; x12
x2 -&gt; K1 -&gt; x21
Compare x11 and x12
Compare x21 and x22
Take average of the two comparisons</p>
</p></li>
<li><p><strong>num_cores</strong> (<em>int</em>) – Number of pools to use for parallel processing across the windows of a chromosome</p></li>
<li><p><strong>param_str</strong> (<em>str</em>) – Parameter string that is the subfolder name under output_dir</p></li>
</ul>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>OrderedDict containing unweighted scores</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chia_rep.check_results">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chia_rep.</span></span><span class="sig-name descname"><span class="pre">check_results</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">rep</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">non_rep</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_file_dir</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">desc_str</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#hichiarep.chia_rep.check_results" title="Link to this definition">¶</a></dt>
<dd><p>Deprecated:
Outputs results in readable text file table format</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>rep</strong> (<em>dict</em><em>(</em><em>str</em><em>, </em><em>dict</em><em>)</em>) – Key: Combined comparison name (LHH0048_LHH0054L)
Value: dict containing keys: ‘emd_value’ and/or ‘j_value’
Dictionary containing information for replicate comparisons</p></li>
<li><p><strong>non_rep</strong> (<em>dict</em><em>(</em><em>str</em><em>, </em><em>dict</em><em>)</em>) – Key: Combined comparison name (LHH0048_LHH0054L)
Value: dict containing keys: ‘emd_value’ and/or ‘j_value’
Dictionary containing information for non-replicate comparisons</p></li>
<li><p><strong>out_file_dir</strong> (<em>str</em><em>, </em><em>optional</em>) – Default is None</p></li>
<li><p><strong>desc_str</strong> (<em>str</em><em>, </em><em>optional</em>) – Used as file name to describe settings/parameters for this comparison
Not optional if out_file_dir is not None
Default is None</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chia_rep.read_data">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chia_rep.</span></span><span class="sig-name descname"><span class="pre">read_data</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">input_data_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom_size_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chroms_to_load</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">List</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">3000000</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">window_stride</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">2</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">num_peaks</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span><span class="w"> </span><span class="p"><span class="pre">|</span></span><span class="w"> </span><span class="pre">None</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">base_chrom</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'chr1'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">normalization</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'NONE'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">bin_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'output'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_hic_value</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_bedgraph_value</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ba_mult</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><a class="reference internal" href="#hichiarep.genome_bin_data.GenomeBinData" title="hichiarep.genome_bin_data.GenomeBinData"><span class="pre">GenomeBinData</span></a><span class="p"><span class="pre">]</span></span></span></span><a class="headerlink" href="#hichiarep.chia_rep.read_data" title="Link to this definition">¶</a></dt>
<dd><p>Reads all samples that are found in loop_data_dir.</p>
<p>loop_data_dir/peak_data_dir/bedgraph_data_dir do not have to be separate
directories.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>input_data_file</strong> (<em>str</em>) – File with file paths to all necessary input files.
Format:
sample1_name bedgraph1_file   hic1_file
sample2_name bedgraph2_file   hic2_file
…</p></li>
<li><p><strong>chrom_size_file</strong> (<em>str</em>) – Path to chromosome size file</p></li>
<li><p><strong>chroms_to_load</strong> (<em>list</em><em>, </em><em>optional</em>) – Specify specific chromosomes to load instead of the entire genome</p></li>
<li><p><strong>window_size</strong> (<em>int</em>) – Size of sliding window</p></li>
<li><p><strong>window_stride</strong> (<em>int</em>) – Factor of window_size to stride (e.g. 2 means stride by window_size/2)</p></li>
<li><p><strong>num_peaks</strong> (<em>int</em><em>, </em><em>optional</em>) – Number of peaks to keep in <cite>base_chrom</cite>
The same ratio of peaks will be kept for other chromosomes
If None, keep all peaks</p></li>
<li><p><strong>base_chrom</strong> (<em>str</em>) – Chromosome to use when selecting number of peaks</p></li>
<li><p><strong>normalization</strong> (<em>str</em><em>, </em><em>optional</em>) – Hi-C normalization method.
Options are same as .hic file normalization methods (e.g. NONE, VC, VC_SQRT, KR)
WARNING: If the normalization results in values that are less than 1,
then those values will be filtered out by two mechanisms:
(1) min_hic_value
(2) chrom_bin_data.py: <cite>laplacian</cite> function which considers values ≥ 1</p></li>
<li><p><strong>bin_size</strong> (<em>int</em>) – Binning resolution</p></li>
<li><p><strong>min_hic_value</strong> (<em>int</em>) – Minimum Hi-C value to consider a valid interaction</p></li>
<li><p><strong>min_bedgraph_value</strong> (<em>int</em>) – Minimum bedgraph value to consider a valid binding affinity</p></li>
<li><p><strong>output_dir</strong> (<em>str</em>) – Directory to output data</p></li>
<li><p><strong>ba_mult</strong> (<em>int</em>) – Multiplicative factor to multiply binding affinity values by
This is recommended if there are binding affinity values are less than 1
(i.e. non-integer values between 0 and 1)</p></li>
</ul>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>OrderedDict[str, <a class="reference internal" href="#hichiarep.genome_bin_data.GenomeBinData" title="hichiarep.genome_bin_data.GenomeBinData">GenomeBinData</a>]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chia_rep.preprocess">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chia_rep.</span></span><span class="sig-name descname"><span class="pre">preprocess</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">sample_dict</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">OrderedDict</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">both_peak_support</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_dir</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">'output'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">diagnostic_plots</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">bool</span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">None</span></span></span><a class="headerlink" href="#hichiarep.chia_rep.preprocess" title="Link to this definition">¶</a></dt>
<dd><p>Filters out Hi-C contacts without peak support using matrix operations</p>
<p>Removes chromosomes that result in all zero matrices after filtering</p>
<p>Optionally saves diagnostic plots of peak array</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>sample_dict</strong> (<em>OrderedDict</em><em>[</em><em>str</em><em>, </em><em>GenomeLoopData</em><em>]</em>) – Samples to compare</p></li>
<li><p><strong>both_peak_support</strong> (<em>bool</em><em>, </em><em>optional</em>) – Whether to only keep loops that have peak support on both sides</p></li>
<li><p><strong>output_dir</strong> (<em>str</em><em>, </em><em>optional</em>) – Directory to output found peaks and filters</p></li>
<li><p><strong>diagnostic_plots</strong> (<em>bool</em><em>, </em><em>optional</em>) – Whether to save diagnostic plots of peak array</p></li>
</ul>
</dd>
<dt class="field-even">Return type<span class="colon">:</span></dt>
<dd class="field-even"><p>None</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chia_rep.read_peak_file">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chia_rep.</span></span><span class="sig-name descname"><span class="pre">read_peak_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">peak_file_path</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">list</span><span class="p"><span class="pre">]</span></span></span></span><a class="headerlink" href="#hichiarep.chia_rep.read_peak_file" title="Link to this definition">¶</a></dt>
<dd><p>Finds the start and ends of every peak in chromosome for one sample. Find
the max value within the interval for each peak using the bedgraph file.</p>
<p>File format must have at least 3 columns with the first 3 being:
chrom_name   start   end</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>peak_file_path</strong> (<em>str</em>) – File path of peak file</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Dictionary containing list of peaks start and ends for every chromosome</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict[str, list]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chia_rep.generate_peak_file">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chia_rep.</span></span><span class="sig-name descname"><span class="pre">generate_peak_file</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">bin_size</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">int</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chrom_size_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">list</span><span class="p"><span class="pre">]</span></span></span></span><a class="headerlink" href="#hichiarep.chia_rep.generate_peak_file" title="Link to this definition">¶</a></dt>
<dd><p>Generates peak dictionary of same format as read_peak_file with
evenly spaced peaks across all chromosomes</p>
<p>This function calls pybedtools window_maker which is a wrapper for
bedtools makewindows to construct this peak dictionary</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>bin_size</strong> (<em>int</em>) – The resolution or window size</p></li>
<li><p><strong>chrom_size_file</strong> (<em>str</em>) – Chromosome size file</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Dictionary containing list of peaks start and ends for every chromosome</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict[str, list]</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="hichiarep.chia_rep.construct_empty_peak_dict">
<span class="sig-prename descclassname"><span class="pre">hichiarep.chia_rep.</span></span><span class="sig-name descname"><span class="pre">construct_empty_peak_dict</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">chrom_size_file</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">Dict</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">,</span></span><span class="w"> </span><span class="pre">list</span><span class="p"><span class="pre">]</span></span></span></span><a class="headerlink" href="#hichiarep.chia_rep.construct_empty_peak_dict" title="Link to this definition">¶</a></dt>
<dd><p>Constructs an empty peak dictionary of same format as read_peak_file with
no peaks for any chromosome</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>chrom_size_file</strong> (<em>str</em>) – Chromosome size file</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Dictionary containing empty list of peaks for every chromosome</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>dict[str, list]</p>
</dd>
</dl>
</dd></dl>

</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="Main">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">HiChIA-rep</a></h1>









<search id="searchbox" style="display: none" role="search">
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false" placeholder="Search"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</search>
<script>document.getElementById('searchbox').style.display = "block"</script><h3>Navigation</h3>
<p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">hichiarep package</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#module-hichiarep.chrom_bin_data">hichiarep.chrom_bin_data module</a></li>
<li class="toctree-l2"><a class="reference internal" href="#module-hichiarep.genome_bin_data">hichiarep.genome_bin_data module</a></li>
<li class="toctree-l2"><a class="reference internal" href="#module-hichiarep.util">hichiarep.util module</a></li>
<li class="toctree-l2"><a class="reference internal" href="#module-hichiarep.chia_rep">hichiarep.chia_rep module</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="index.html" title="previous chapter">Welcome to HiChIA-Rep’s documentation!</a></li>
  </ul></li>
</ul>
</div>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &#169;2025, Minji Kim, Sion Kim, Henry Zhang.
      
      |
      Powered by <a href="https://www.sphinx-doc.org/">Sphinx 8.1.3</a>
      &amp; <a href="https://alabaster.readthedocs.io">Alabaster 1.0.0</a>
      
      |
      <a href="_sources/hichiarep.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>